corehq/util/es/elasticsearch.py

```diff
@@ -0,0 +1,60 @@
+from django.conf import settings
+
+if settings.ELASTICSEARCH_MAJOR_VERSION == 1:
+    import elasticsearch
+    from elasticsearch.exceptions import AuthorizationException
+    from elasticsearch import (
+        ConnectionError,
+        ConnectionTimeout,
+        Elasticsearch,
+        ElasticsearchException,
+        NotFoundError,
+        SerializationError,
+        ConflictError,
+        TransportError,
+        RequestError,
+    )
+    from elasticsearch.client import (
+        IndicesClient,
+        SnapshotClient,
+    )
+    from elasticsearch.helpers import bulk
+elif settings.ELASTICSEARCH_MAJOR_VERSION == 2:
+    import elasticsearch2 as elasticsearch
+    from elasticsearch2.exceptions import AuthorizationException
+    from elasticsearch2 import (
+        ConnectionError,
+        ConflictError,
+        ConnectionTimeout,
+        Elasticsearch,
+        ElasticsearchException,
+        NotFoundError,
+        SerializationError,
+        TransportError,
+        RequestError,
+    )
+    from elasticsearch2.client import (
+        IndicesClient,
+        SnapshotClient,
+    )
+    from elasticsearch2.helpers import bulk
+else:
+    raise ValueError("ELASTICSEARCH_MAJOR_VERSION must currently be 1 or 2")
+
+
+__all__ = [
+    'AuthorizationException',
+    'ConflictError',
+    'ConnectionError',
+    'ConnectionTimeout',
+    'Elasticsearch',
+    'ElasticsearchException',
+    'IndicesClient',
+    'NotFoundError',
+    'RequestError',
+    'SerializationError',
+    'SnapshotClient',
+    'TransportError',
+    'bulk',
+    'elasticsearch',
+]
```

Comment:
Do you have further plans for `corehq/util/es/` beyond this file?  OOC, why put this all here rather than in `corehq/util/es/__init__.py` or `corehq/util/es.py`?

Reply:
I do, though they aren't fully fleshed out. As a next step, I want to write wrappers for specific calls that have different behavior on ES 1 and ES 2, and have them in this same package in a different module. So `corehq.util.es.elasticsearch` is solely for replacing `from elasticsearch...` imports so it always imports from the right version of the ES library according to your ES version, and the other module or modules under `corehq.util.es` will be for these other wrappers for further removing the difference between actual ES version being used. (These wrappers themselves will probably also import from `corehq.util.es.elasticsearch`.)

Reply:
An example of how ES 1 and ES 2 behave differently (which the `elasticsearch` library just passed through to its own interface) is that you can't include `_id` in docs anymore as it's a protected field. The `_id` field still exists but it's set through the API and not as part of the doc source anymore.